cluster/swarm/cluster.go

```diff
@@ -359,9 +359,13 @@ func (c *Cluster) removeEngine(addr string) bool {
 func (c *Cluster) monitorDiscovery(ch <-chan discovery.Entries, errCh <-chan error) {
 	// Watch changes on the discovery channel.
 	currentEntries := discovery.Entries{}
-	for {
+	for ch != nil || errCh != nil {
 		select {
-		case entries := <-ch:
+		case entries, ok := <-ch:
+			if !ok {
+				ch = nil
```

Comment:
@dongluochen  
Sorry, I still can't get your point. The framework of `monitorDiscovery` is like this:  

	func (c *Cluster) monitorDiscovery(ch <-chan discovery.Entries, errCh <-chan error) {
		// Watch changes on the discovery channel.
		currentEntries := discovery.Entries{}
		for {
			select {
			case entries := <-ch:
				......				
			case err := <-errCh:
				......
			}
		}
	}
(1) The channels `ch` and `errCh` are parameters of `monitorDiscovery`, and if they are closed at other places, I don't think it is meaningful of still receiving values from these channels;  
(2) For the already closed channels `ch` and `errCh`, they are only visible in `monitorDiscovery`. If setting them values as `nil`, what are the side-effects?

Thanks!